metrics/web/lib/dashboard/domain/usecases/receive_build_day_project_metrics_updates.dart

```diff
@@ -80,50 +89,32 @@ class ReceiveBuildDayProjectMetricsUpdates
     );
   }
 
-  /// Calculates the average [Duration] of [buildDays] using the
-  /// [BuildDay.totalDuration].
-  ///
-  /// Returns a [Duration.zero] if there are no builds performed during the
+  /// Calculates the [PerformanceMetric.averageBuildDuration] based on the
   /// given [buildDays].
+  ///
+  /// Returns a [Duration.zero] if there are no successful builds performed
+  /// during the given [buildDays].
   Duration _calculateAverageDuration(List<BuildDay> buildDays) {
-    final numberOfBuildsInPeriod = _calculateTotalNumberOfBuilds(buildDays);
-
-    if (numberOfBuildsInPeriod == 0) return Duration.zero;
-
-    final totalDurationInPeriod = _calculateTotalDurationOfBuilds(buildDays);
-
-    return totalDurationInPeriod ~/ numberOfBuildsInPeriod;
-  }
-
-  /// Returns the total number of builds performed during the given [buildDays].
-  int _calculateTotalNumberOfBuilds(List<BuildDay> buildDays) {
-    return buildDays.fold<int>(
+    final numberOfSuccessfulBuilds = buildDays.fold<int>(
```

Comment:
Would it be better to calculate the `numberOfSuccessfulBuilds` and `successfulBuildsDuration` in one `for each` loop? What do you think?